graph/vertex/impl/ValueVertexImpl.java

```diff
@@ -0,0 +1,308 @@
+        }
+
+        public Encoding.ValueType<java.lang.Double> valueType() {
+            return Encoding.ValueType.DOUBLE;
+        }
+
+        @Override
+        public boolean isDouble() {
+            return true;
+        }
+
+        @Override
+        public Double asDouble() {
+            return this;
+        }
+
+    }
+
+    public static class String extends ValueVertexImpl<java.lang.String> {
+
+        String(VertexIID.Value<java.lang.String> iid, java.lang.String value) {
+            super(iid, value);
+        }
+
+        public Encoding.ValueType<java.lang.String> valueType() {
+            return Encoding.ValueType.STRING;
+        }
+
+        @Override
+        public boolean isString() {
+            return true;
+        }
+
+        @Override
+        public String asString() {
+            return this;
+        }
+
+    }
+
+    public static class DateTime extends ValueVertexImpl<LocalDateTime> {
+
+        DateTime(VertexIID.Value<LocalDateTime> iid, LocalDateTime value) {
+            super(iid, value);
+        }
+
+        public Encoding.ValueType<LocalDateTime> valueType() {
+            return Encoding.ValueType.DATETIME;
+        }
+
+        @Override
+        public boolean isDateTime() {
+            return true;
+        }
+
+        @Override
+        public DateTime asDateTime() {
+            return this;
+        }
+
+    }
+
+    public static class ValueVertexSpec<T> extends ValueVertexImpl<T> implements ValueVertex.ValueVertexSpec<T> {
+        private final ValueHolder<T> valueVertex;
+
+        public ValueVertexSpec(VertexIID.Value<T> iid, ValueHolder<T> valueVertex) {
+            super(iid, valueVertex.value());
+            this.valueVertex = valueVertex;
+        }
+
+        public static ValueVertexImpl<?> of(ValueHolder<?> valueVertex) {
+            if (valueVertex.isBoolean())
+                return new ValueVertexImpl.ValueVertexSpec<>(new VertexIID.Value.Boolean(valueVertex.asBoolean().value()), valueVertex.asBoolean());
+            else if (valueVertex.isLong())
+                return new ValueVertexImpl.ValueVertexSpec<>(new VertexIID.Value.Long(valueVertex.asLong().value()), valueVertex.asLong());
+            else if (valueVertex.isDouble())
+                return new ValueVertexImpl.ValueVertexSpec<>(new VertexIID.Value.Double(valueVertex.asDouble().value()), valueVertex.asDouble());
+            else if (valueVertex.isString()) {
+                try { return new ValueVertexImpl.ValueVertexSpec<>(new VertexIID.Value.String(valueVertex.asString().value()), valueVertex.asString()); }
+                catch (TypeDBCheckedException e) { throw TypeDBException.of(ILLEGAL_STATE, e); }
+            } else if (valueVertex.isDateTime())
+                return new ValueVertexImpl.ValueVertexSpec<>(new VertexIID.Value.DateTime(valueVertex.asDateTime().value()), valueVertex.asDateTime());
+            else throw TypeDBException.of(ILLEGAL_STATE);
+        }
```

Comment:
This approach of assigning IIDs doesn't really make sense. 
Should I introduce a new class of VertexIIDs for VertexSpecs? Or we rethinking VertexSpecs altogether.